client/app/lib/providers/connectcompute.coffee

```diff
@@ -0,0 +1,88 @@
+React = require 'app/react'
+kd = require 'kd'
+shallowCompare = require 'react-addons-shallow-compare'
+
+# FIXME: so naive, but works for now.
+makeSingular = (plural) -> plural.slice 0, -1
+
+
+makeState = (config, props) ->
+
+  return {}  unless config.require
+
+  { storage } = kd.singletons.computeController
```

Comment:
there might be cases that this statement needs to be accessible before `computeController` is created/ready. did you take that into account?